File: Examples/DotNet/CSharp/Oleobjects/AddOleObjectsWorksheet.cs
using System;
using Com.Aspose.Cells.Api;
using Com.Aspose.Cells.Model;
using Com.Aspose.Storage.Api;

namespace OleObjects
{
    class AddOleObjectsWorksheet
    {
        public static void Run()
        {
            // ExStart:1
            CellsApi cellsApi = new CellsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            string fileName = "Sample_Test_Book.xls";
            string sourceFileName = "Sample_Book2.xls";
            string imageFileName = "aspose-logo.png";

            string sheetName = "Sheet1";
            int? upperLeftRow = null;
            int? upperLeftColumn = null;
            int? height = null;
            int? width = null;
            string oleFile = null;
            string imageFile = null;
            string storage = null;
            string folder = null;
            Com.Aspose.Cells.Model.OleObject body = new Com.Aspose.Cells.Model.OleObject();
            body.SourceFullName = sourceFileName;
            body.ImageSourceFullName = imageFileName;
            body.UpperLeftRow = 15;
            body.UpperLeftColumn = 5;
            body.Top = 10;
            body.Bottom = 10;
            body.Left = 10;
            body.Height = 400;
            body.Width = 400;
            body.IsAutoSize = true;

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));
                storageApi.PutCreate(sourceFileName, null, null, System.IO.File.ReadAllBytes(Common.GetDataDir() + sourceFileName));
                storageApi.PutCreate(imageFileName, null, null, System.IO.File.ReadAllBytes(Common.GetDataDir() + imageFileName));

                // Invoke Aspose.Cells Cloud SDK API to add OleObject
                OleObjectResponse apiResponse = cellsApi.PutWorksheetOleObject(fileName, sheetName, upperLeftRow, upperLeftColumn, height, width, oleFile, imageFile, storage, folder, body);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    Console.WriteLine("Add OleObjects to Excel Worksheet, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Perl/Oleobjects/AddOleObjectsWorksheet.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;
use AsposeCellsCloud::CellsApi;
use AsposeCellsCloud::ApiClient;
use AsposeCellsCloud::Configuration;
use  AsposeCellsCloud::Object::OleObject;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeCellsCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeCellsCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeCellsCloud::Configuration::debug = 1;
$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Cells API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $cellsApi = AsposeCellsCloud::CellsApi->new();

# Set input file name
my $name = 'Sample_Test_Book.xls';
my $sheetName = 'Sheet1';
my $sourceFileName = 'Sample_Book2.xls';
my $imageFileName = 'aspose-logo.png';
my @oleObjectBody = AsposeCellsCloud::Object::OleObject->new(
'SourceFullName' => $sourceFileName,		
'ImageSourceFullName' => $imageFileName,
'UpperLeftRow' => 15,
'UpperLeftColumn' => 5,
'Top' => 10,
'Bottom' => 10,
'Left' => 10,
'Height' => 400,
'Width' => 400,
'IsAutoSize' => 'True'
);

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);
$response = $storageApi->PutCreate(Path => $sourceFileName, file => $data_path.$sourceFileName);
$response = $storageApi->PutCreate(Path => $imageFileName, file => $data_path.$imageFileName);

# Invoke Aspose.Cells Cloud SDK API to add an OleObject to a worksheet                   
$response = $cellsApi->PutWorksheetOleObject(name=> $name, sheetName=>$sheetName, body=>@oleObjectBody);

if($response->{'Status'} eq 'OK'){
	# Download updated Workbook from storage server 
	my $output_file =  $out_path. $name;
	$response = $storageApi->GetDownload(Path => $name);;
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1